helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -75,6 +75,35 @@ receivers:
           endpoint: '`endpoint`:`"prometheus.io/port" in annotations ? annotations["prometheus.io/port"] : 9090`'
       {{- end }}
 
+      # Receivers for collecting k8s control plane metrics.
+      {{- if .Values.autodetect.k8sControlPlane }}
+      smartagent/kubernetes-apiserver:
+        # Except for openshift, Kubernetes distributions typically use the same apiserver pod labels.
+        {{- if eq (include "splunk-otel-collector.distribution" .) "openshift" }}
+        rule: type == "pod" && labels["app"] == "openshift-kube-apiserver" && labels["apiserver"] == "true"
+        {{- else }}
+        rule: type == "pod" && labels["k8s-app"] == "kube-apiserver"
+        {{- end }}
+        config:
+          # If the apiserver config is defined by the user, then use that config here.
+          {{- if .Values.k8sControlPlane.apiserver.config -}}
+          - {{- toYaml $.Values.k8sControlPlane.apiserver.config | nindent 10 }}
+          {{- else }}
+          # If the apiserver config is not defined, then we use this default config.
+          - host: localhost
+          # Except for openshift, Kubernetes distributions typically use the same apiserver port.
+          {{- if eq (include "splunk-otel-collector.distribution" .) "openshift" }}
+          - port: 6443
+          {{- else }}
+          - port: 443
+          {{- end }}
+          {{- end }}
+          # We skip verifying here because the k8s default certificate is self signed and will fail this verification.
+          - skipVerify: true
+          - useHTTPS: true
+          - useServiceAccount: true
```

Comment:
I think all these values should not be a part of a list.

Reply:
Fixed this.